packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -303,6 +303,18 @@ export enum Effect {
   DENY = 'Deny',
 }
 
+/**
+ * Conditions for when an IAM Policy is in effect, specified in the following structure:
+ *
+ * `{ "Operator": { "keyInRequestContext": "value" } }`
+ *
+ * The value can be either a single string value or an array of string values.
+ *
+ * For more information, including which operators are supported, see [the IAM
+ * documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition.html).
+ */
+export type Conditions = Record<string, any>;
```

Comment:
Wait I believe we said two-level types here, right?

```ts
type Conditions = Record<string, Condition>;
type Condition = Record<string, any>;
```

Is the compatibility checker giving you guff if you change this?

Reply:
Yeah, I tried it and it still flagged as causing breaking changes. 

From [the failed build logs](https://6sc75qyfl7.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=aab5f570-da0b-4efe-8400-b9e2aab151ad%2Fbuild.log):

```
API elements with incompatible changes:
err  - INITIALIZER @aws-cdk/aws-iam.FederatedPrincipal.<initializer>: argument conditions, takes Map<string => Map<string => any>> (formerly Map<string => any>): Map<string => any> is not assignable to Map<string => Map<string => any>>, any is not a map type [incompatible-argument:@aws-cdk/aws-iam.FederatedPrincipal.<initializer>]
err  - METHOD @aws-cdk/aws-iam.PolicyStatement.addCondition: argument value, takes Map<string => string | Array<string>> (formerly any): any is not a map type [incompatible-argument:@aws-cdk/aws-iam.PolicyStatement.addCondition]
err  - METHOD @aws-cdk/aws-iam.PolicyStatement.addConditions: argument conditions, takes Map<string => Map<string => any>> (formerly Map<string => any>): Map<string => any> is not assignable to Map<string => Map<string => any>>, any is not a map type [incompatible-argument:@aws-cdk/aws-iam.PolicyStatement.addConditions]
err  - METHOD @aws-cdk/aws-iam.PolicyStatement.addFederatedPrincipal: argument conditions, takes Map<string => Map<string => any>> (formerly Map<string => any>): Map<string => any> is not assignable to Map<string => Map<string => any>>, any is not a map type [incompatible-argument:@aws-cdk/aws-iam.PolicyStatement.addFederatedPrincipal]
err  - INITIALIZER @aws-cdk/aws-iam.PrincipalPolicyFragment.<initializer>: argument conditions, takes Optional<Map<string => Map<string => any>>> (formerly Optional<Map<string => any>>): Map<string => any> is not assignable to Map<string => Map<string => any>>, any is not a map type [incompatible-argument:@aws-cdk/aws-iam.PrincipalPolicyFragment.<initializer>]
```

For reference, here's [the commit](https://github.com/aws/aws-cdk/pull/7015/commits/ffc2bdc035d8a3969891affa4662a975a256247f) where I changed it back (and caught the union type I'd somehow missed removing before).